src/components/forms/validator.ts

```diff
@@ -2,20 +2,14 @@ import { List } from 'immutable'
 
 import { sameAddress } from 'src/logic/wallets/ethAddresses'
 import { getWeb3 } from 'src/logic/wallets/getWeb3'
+import memoize from 'lodash.memoize'
 
-export const simpleMemoize = (fn) => {
-  let lastArg
-  let lastResult
-  return (arg, ...args) => {
-    if (arg !== lastArg) {
-      lastArg = arg
-      lastResult = fn(arg, ...args)
-    }
-    return lastResult
-  }
-}
+type ValidatorReturnType = string | undefined
+type GenericValidatorType = (...args: unknown[]) => ValidatorReturnType
+type AsyncValidator = (...args: unknown[]) => Promise<ValidatorReturnType>
+type Validator = GenericValidatorType | AsyncValidator
```

Comment:
just curious, why are we defining these types instead of using the `final-form` ones? https://github.com/final-form/final-form/blob/master/src/index.d.ts#L131

Reply:
🤦‍♂️

Reply:
I think ours is more strict, for final-form ones they return any and don't provide a way to specify return type. Let's go with those